extensions/cornerstone-dynamic-volume/src/panels/PanelGenerateImage.tsx

```diff
@@ -0,0 +1,414 @@
+import React, { useEffect, useState, useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+import PropTypes, { array } from 'prop-types';
+import { Button, Select, InputDoubleRange, Label, Icon } from '@ohif/ui';
+import { useCine, useViewportGrid } from '@ohif/ui';
+import { cache, utilities as csUtils, volumeLoader } from '@cornerstonejs/core';
+import {
+  CONSTANTS as cstConstants,
+  Enums as csToolsEnums,
+  segmentation as cstSegmentation,
+  Types as cstTypes,
+  utilities as cstUtils,
+} from '@cornerstonejs/tools';
+
+const DEFAULT_OPTIONS = {
+  TimeFrames: null,
+  Operation: 'SUM',
+};
+
+const SUM = 'SUM';
+const AVG = 'AVERAGE';
+const SUB = 'SUBTRACT';
+
+const operations = [
+  { value: SUM, label: 'SUM', placeHolder: 'SUM' },
+  { value: AVG, label: 'AVERAGE', placeHolder: 'AVERAGE' },
+  { value: SUB, label: 'SUBTRACT', placeHolder: 'SUBTRACT' },
+];
+
+export default function PanelGenerateImage({
+  servicesManager,
+  commandsManager,
+}) {
+  const {
+    viewportGridService,
+    cornerstoneViewportService,
+    displaySetService,
+    hangingProtocolService,
+  } = servicesManager.services;
+  const { t } = useTranslation('PanelGenerateImage');
+  const [generateOptions, setGenerateOptions] = useState(DEFAULT_OPTIONS);
+  const [timeOptions, setTimeOptions] = useState([]);
+  const [rangeValues, setRangeValues] = useState([]);
+  const [sliderValues, setSliderValues] = useState([]);
```

Comment:
I don't see `sliderValues` getting used anywhere, so I'm wondering why you need a state here